sdk/textanalytics/ai-text-analytics/src/subscriptionKeyCredential.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { ServiceClientCredentials, WebResource } from "@azure/core-http";
+
+const SUBSCRIPTION_KEY_HEADER_NAME = "Ocp-Apim-Subscription-Key";
+
+/**
+ * This class is used to authenticate to Text Analytics using a Subscription Key retrieved
+ * from the Azure portal.
+ */
+export class SubscriptionKeyCredential implements ServiceClientCredentials {
```

Comment:
Just wondering why not follow the [SDK guidelines](https://github.com/Azure/azure-sdk/issues/947) that say it needs have the service name for this?

Reply:
Sorry, there was some discussion this morning. 

My concern is the name was getting too long and for JS we don't have to worry about name collisions since our types don't exist at runtime. We can talk more on Friday though! :)

Reply:
Thanks!  Looking forward to learning more. :)